store/index-refactor.js

```diff
@@ -122,6 +122,11 @@ export const getters = {
 
 export const actions = {
 
+    processDataDictionary( { state, commit, getters }, { data, filename }) {
+
+        commit("setDataDictionary", data, getters.getColumnNames);
```

Comment:
This won't work because the `fileSelector` component returns a string, not an object. See [here](https://github.com/neurobagel/annotation_tool/blob/dev_components_talk_to_store_directly/components/file-selector.vue#L105) and [here](https://github.com/neurobagel/annotation_tool/blob/dev_components_talk_to_store_directly/cypress/component/index_page.cy.js#L126).

I think this in itself might be enough reason to fix up the `fileSelector` component to instead output an object (because that's what you'd expect).

Your call. Either we can parse string -> object in the action and then later refactor the fileSelector component, or we can do it here.

Reply:
Let's keep the PR as small as possible as we've mentioned. When fileSelector is refactored and that's the focus we can make the change so that it returns an object.

In the meantime, I'll go with `JSON.parse(data)` here.